verticapy/pipeline/_transform.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python3
+"""
+Copyright  (c)  2018-2024 Open Text  or  one  of its
+affiliates.  Licensed  under  the   Apache  License,
+Version 2.0 (the  "License"); You  may  not use this
+file except in compliance with the License.
+
+You may obtain a copy of the License at:
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless  required  by applicable  law or  agreed to in
+    writing, software  distributed  under the  License is
+    distributed on an  "AS IS" BASIS,  WITHOUT WARRANTIES
+    OR CONDITIONS OF ANY KIND, either express or implied.
+    See the  License for the specific  language governing
+    permissions and limitations under the License.
+"""
+"""
+This script runs the Vertica Machine Learning Pipeline Transforming.
+"""
+
+
+from queue import Queue
+
+import verticapy as vp
+from verticapy.core.vdataframe.base import vDataFrame
+
+
+def reset_queue(column_queue: Queue) -> Queue:
+    """
+    Return a new Queue where all columns haven't been tried.
+    """
+    new_queue = Queue()
+    while not column_queue.empty():
+        col, _ = column_queue.get()
+        new_queue.put((col, False))
+    return new_queue
+
+
+def transformation(transform: dict, table: str) -> vDataFrame:
+    """
+    Run the transformation step
+    of the pipeline.
+
+    Parameters
+    ----------
+    transform: dict
+        YAML object which outlines the steps of the operation.
+    table: str
+        The name of the table the pipeline is ingesting to.
+
+    Returns
+    -------
+    vDataFrame
+        The transformed vDataFrame.
+    """
+    vdf = vp.vDataFrame(table)
+
+    column_queue = Queue()
+    for col in sorted(list(transform.keys())):
+        column_queue.put((col, False))
+
+    error_string = ""
+    while not column_queue.empty():
+        col, is_dep = column_queue.get()
+        if is_dep:
+            raise RuntimeError(
+                "Error: All remaining columns have dependencies\n" + error_string
```

Comment:
I think it would be nice to let the user know which column has a dependency, how do you know all the rest do?

Reply:
The error_string should have all the columns it had trouble creating ie if there was three columns a, b, c and they all used each other like a = b and b = c and c = a, then it would give an error along the lines: 
a can't be created because b doesn't exist
b can't be created because c doesn't exist
c can't be created because a doesn't exist

maybe a better error message would be "Error: All remaining columns either have an error or cyclic dependencies."